### DISCUSSION TITLE
Project-Origin topics for Software Engineering Thesis

### DISCUSSION THREAD

**Discussion by lauranolling** [project-origin/registry#97] [Q&A]
# Project-Origin topics for Software Engineering Thesis

One of the next steps agreed in [our last workshop 01.03.2023](https://github.com/project-origin/registry/blob/main/meeting_minutes/workshop_01_03_2023.md) was to "Invite university partnerships" to participate in Project-Origin. Coincidentally, Software Engineering students at SDU Odense is starting to choose projects for their 40 ECTS Master Thesis. 

In this context I want to make a short Project Description for them, to promote Project-Origin amongst the students, and hopefully attract one/some of them to participate in the implementation here as part of their master thesis. 

## The Question
I need some help defining exactly: **Which bigger implementation tasks are ahead in Project-Origin?** 

Identifying such topics/tasks will help write a more motivating project description. 

I have looked at the [Project Roadmap](https://project-origin.github.io/registry/roadmap.html) for the _registry_ implementation, but the checklists for the "alpha" version, "beta" version and "Long term goals" don't reveal much about the size and complexity of the different tasks ahead. Can someone please help clarify that?

**Comment by wisbech** [comment]
Hi Laura.

Excellent initiative indeed - we have 2 major hurdles ahead of us for project origin maybe even 4:

1. Audit feature - any consumer or 3. party service should be able to validate their own claims and the registries should be able to proof that they have not been tampered with since publication of the hash to a given blockchain. 
2. Scalability and performance improvements of project-origin as it is today. 
    - Benchmarking.
    - Pure rust implementation improvement ? 
    - Federated Network setup - we are currently looking at [hyperledger firefly](https://github.com/hyperledger/firefly) but other similar implementations could be of interest.
4. Simulation of an energy eco-system that uses project origin for digital proofs and validation, ie. digital twin infrastructure etc.

**Comment by lauranolling** [comment]
Thank you for these inputs @wisbech. I will continue work on a project description, and include these as potential topics.
